File: lint/src/main/java/com/firebaseui/lint/LintIssueRegistry.java
package com.firebaseui.lint;

import com.android.tools.lint.client.api.IssueRegistry;
import com.android.tools.lint.detector.api.Issue;

import java.util.Collections;
import java.util.List;

/**
 * Registry for custom FirebaseUI lint checks.
 */
public class LintIssueRegistry extends IssueRegistry {

    @Override
    public List<Issue> getIssues() {
        return Collections.singletonList(NonGlobalIdDetector.NON_GLOBAL_ID);
    }

}

File: lint/src/main/java/com/firebaseui/lint/LintIssueRegistry.kt
package com.firebaseui.lint

import com.android.tools.lint.client.api.IssueRegistry
import com.android.tools.lint.client.api.Vendor

/**
 * Registry for custom FirebaseUI lint checks.
 */
class LintIssueRegistry : IssueRegistry() {
    override val api: Int
        get() = com.android.tools.lint.detector.api.CURRENT_API

    override val issues = listOf(
            FirestoreRecyclerAdapterLifecycleDetector.ISSUE_MISSING_LISTENING_START_METHOD,
            FirestoreRecyclerAdapterLifecycleDetector.ISSUE_MISSING_LISTENING_STOP_METHOD
    )

    override val vendor = Vendor(
        vendorName = "FirebaseUI Android",
        identifier = "com.firebaseui.lint",
        feedbackUrl = "https://github.com/firebase/FirebaseUI-Android",
        contact = "https://github.com/firebase/FirebaseUI-Android"
    )
}
